fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBInQueryTest.java

```diff
@@ -152,20 +168,20 @@ void testInQueryNoIndex() throws Exception {
         } else {
             assertMatchesExactly(plan,
                     predicatesFilterPlan(descendantPlans(scanPlan().where(scanComparisons(unbounded()))))
-                            .where(predicates(valuePredicate(ValueMatchers.fieldValueWithFieldNames("num_value_2"), new Comparisons.ListComparison(Comparisons.Type.IN, ImmutableList.of(0, 2))))));
+                            .where(predicates(valuePredicate(fieldValueWithFieldNames("num_value_2"), new Comparisons.ListComparison(Comparisons.Type.IN, ImmutableList.of(0, 2))))));
 
             assertEquals(738091077, plan.planHash(PlanHashable.CURRENT_LEGACY));
             assertEquals(-2062922437, plan.planHash(PlanHashable.CURRENT_FOR_CONTINUATION));
         }
         assertEquals(67, querySimpleRecordStore(NO_HOOK, plan, EvaluationContext::empty,
-                record -> assertThat(record.getNumValue2(), anyOf(is(0), is(2))),
+                rec -> assertThat(rec.getNumValue2(), anyOf(is(0), is(2))),
                 context -> assertDiscardedAtMost(33, context)));
     }
 
     /**
      * Verify that an IN (with parameter) without an index is implemented as a filter on a scan.
      */
-    @Test
+    @DualPlannerTest
```

Comment:
This is the test class with the highest number of changes, so it's worth specifying what changed here:

1. I updated all of the tests to be `DualPlanner` tests so that we'd get coverage of both planners. For many tests, the plans produced are similar, though they may use different implementation classes (e.g., `InValues` joins instead of `InParameter` joins). All of the existing queries already worked with Cascades, so this just served to double check this
2. For queries that specified a sort, I parameterized the query by `reverse` and now assert that the plan's reverse-ness matches the parameter's. The reverse version of the plans have different hashes; for all of the existing plans, the hashes didn't change. I also modified the runtime checks to validate that these results are correctly sorted
3. The most important additional test is `testParameterizedInQuerySortedWithExtraUnorderedColumns` which checks that when there's an in-parameter predicate extra columns in the index, that we can still plan that as an in-union (when the config is set).
4. The new test `testListInQuerySortedWithExtraUnorderedColumns` is similar, but it uses a list value instead of a list parameter, and it can check both translating the IN into an in-union and translating the IN into an OR (and thus planning it as a regular union). As it happens, when the IN is translated into an OR, this already did the right thing as long as you set the config parameter added in #2408. That mode is more-or-less deprecated, though, so this test is intended a nice validation that things work together.